api.go

```diff
@@ -241,18 +241,28 @@ func (h *Headscale) getMapResponse(mKey wgkey.Key, req tailcfg.MapRequest, m Mac
 		DisplayName: m.Namespace.Name,
 	}
 
+	var dnsConfig *tailcfg.DNSConfig
+	if h.cfg.DNSConfig != nil && h.cfg.DNSConfig.Proxied { // if MagicDNS is enabled
+		// TODO(juanfont): We should not be regenerating this all the time
+		// And we should only send the domains of the peers (this own namespace + those from the shared peers)
+		namespaces, err := h.ListNamespaces()
+		if err != nil {
+			return nil, err
+		}
+		dnsConfig = h.cfg.DNSConfig.Clone()
+		for _, ns := range *namespaces {
+			dnsConfig.Domains = append(dnsConfig.Domains, fmt.Sprintf("%s.%s", ns.Name, h.cfg.BaseDomain))
+		}
+	} else {
+		dnsConfig = h.cfg.DNSConfig
+	}
```

Comment:
Can this snippets be made into a function in preparation for fixing that todo?

Sounds like it could belong in `dns.go`

Reply:
I have reviewed the code, and I reckon we just should send the search domain of the current namespace. Shared nodes need to use the FQDN - as per tailscale.com SaaS. Fixed in the latest commit.